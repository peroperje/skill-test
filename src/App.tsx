import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter, Link, Route } from 'react-router-dom';

import './App.css';
import state from './state';
import {
  ROOT_ROUTE,
  LOGIN_ROUTE,
  FILES_ROUTE
} from './utils/route/routesDefinition';
import RouteOnlyNonLogged from './utils/route/RouteOnlyNonLogged';
import Home from './view/Home';
import Files from './view/Files';
import Login from './view/Login';

const App: React.FC = (): JSX.Element => {
  return (
    <Provider store={state}>
      <BrowserRouter>
        <div className="App">
          <header className="App-header">
            <div className="Flex-row">
              <Link to={ROOT_ROUTE}>Home</Link>
              <Link to={FILES_ROUTE}>File Manager</Link>
              <Link to={LOGIN_ROUTE}>Sign in</Link>
            </div>
            <div>
              <Route path={ROOT_ROUTE} exact component={Home} />
              <RouteOnlyNonLogged path={LOGIN_ROUTE} exact component={Login} />
              <Route path={FILES_ROUTE} exact component={Files} />
            </div>
          </header>
        </div>
      </BrowserRouter>
    </Provider>
  );
};

export default App;
