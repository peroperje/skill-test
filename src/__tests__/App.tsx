import React from 'react';
import * as shallow from 'react-test-renderer/shallow';

import App from '../App';

describe('App', () => {
  it('Shallow Snapshot', () => {
    const render = shallow.createRenderer();
    const component = render.render(<App />);
    expect(component).toMatchSnapshot();
  });
});
