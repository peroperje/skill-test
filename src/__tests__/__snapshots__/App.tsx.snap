// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Shallow Snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <BrowserRouter>
    <Memo(Connect(ShouldFetch)) />
    <ForwardRef(WithStyles)>
      <Route
        component={[Function]}
        path="/"
      />
      <ForwardRef(StyledComponent)
        mt={5}
      >
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(ShouldFetch)",
              "type": [Function],
            }
          }
          exact={true}
          path="/"
        />
        <RouteOnlyNonLogged
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(Login)",
              "type": [Function],
            }
          }
          exact={true}
          path="/login"
        />
      </ForwardRef(StyledComponent)>
    </ForwardRef(WithStyles)>
  </BrowserRouter>
</Provider>
`;
